weight.cpp

```diff
@@ -42,6 +43,33 @@ bool is_active(const weight &w)
  return w.is_active();
 }
 
+std::vector<double> convert_to_double
+  (const std::vector<weight> &weights)
+{
+  std::vector<double> double_vector;
+
+  for(size_t i = 0; i != weights.size(); i++)
+    {
+      double_vector.push_back(weights[i].get_weight());
+    }
```

Comment:
Is good to pre-allocate the size of a vector (as we do in the function `response`
```suggestion
  std::vector<double> double_vector(weights.size());
 
 for(size_t i = 0; i != weights.size(); i++)
    {
      double_vector[i] = weights[i].get_weight());
    }